File: ui/libs/commons/src/lib/shared/models/customer.dto.ts
export interface CustomerDto {
  id?: number;
  externalId: string;
  name: string;
  code?: string;
  phoneNumber?: string;
  customerAddresses: CustomerAddressDto[];
  isbtCodeList: CustomerIsbtCodeDto[];
  createDate?: string;
  modificationDate?: string;
  deleteDate?: string;
}

export interface CustomerAddressDto {
  id?: number;
  customerId: number;
  addressType: string;
  city: string;
  contactName: string;
  state: string;
  postalCode: string;
  country: string;
  countryCode: string;
  district: string;
  addressLine1: string;
  addressLine2: string;
}

export interface CustomerIsbtCodeDto {
  id?: number;
  customerId?: number;
  code: string;
  orderNumber: number;
}

File: backend/recoveredplasmashipping/src/main/java/com/arcone/biopro/distribution/recoveredplasmashipping/adapter/in/web/dto/CustomerDTO.java
package com.arcone.biopro.distribution.recoveredplasmashipping.adapter.in.web.dto;

import lombok.Builder;

import java.io.Serializable;
import java.time.ZonedDateTime;

@Builder
public record CustomerDTO (
    Long id,
    String externalId,
    String customerType,
    String name,
    String code,
    String departmentCode,
    String departmentName,
    String foreignFlag,
    String phoneNumber,
    String contactName,
    String state,
    String postalCode,
    String country,
    String countryCode,
    String city,
    String district,
    String addressLine1,
    String addressLine2,
    Boolean active,
    ZonedDateTime createDate,
    ZonedDateTime modificationDate
) implements Serializable {


}
